torch/distributed/rpc/backend_registry.py

```diff
@@ -267,6 +273,56 @@ def _tensorpipe_check_local_device_maps(name, device_count, options):
             f"device_maps = {options.device_maps},\n"
             f"devices = {options.devices}"
         )
+    return local_devices
+
+def _update_group_membership(worker_info, my_devices, reverse_device_map):
+    agent = api._get_current_rpc_agent()
+    ret = agent._update_group_membership(worker_info, my_devices, reverse_device_map, True)
+    return "hi"
+
+def _get_device_infos():
+    if torch.cuda.is_available():
+        device_count = torch.cuda.device_count()
```

Comment:
`torch.cuda.device_count()` should work even CUDA is not available. 

https://github.com/pytorch/pytorch/blob/cd80ef6cdbc413dbe6a22afd55101168306f7020/torch/cuda/__init__.py#L453-L458